---
title: "17q13q: Clonality Analysis (SITKA TREE)"
output:
  html_document:
    toc: true
    toc_float: true
    theme: united
    code_folding: show
---

# **Editing CNV Data for Sitka Binary Matrix**
## *Luminal Epithelial Data*
### LUMI: Library for Generating Data
```{r message=FALSE, warning=FALSE}
#install.packages("tidyverse")
library(tidyverse)
#install.packages("dplyr")
library(dplyr)
# install.packages("readr")
library(readr)
# install.packages("gt")
library(gt)
# install.packages("vroom")
library(vroom)
#install.packages("openxlsx")
library(openxlsx)
#install.packages("ape")
library(ape)
#install.packages("ggtree")
library(ggtree)
#install.packages("ggplot2")
library(ggplot2)
```

### LUMI: Load CNV Data
```{r}
read_cnv_data <- function(filepath) {
#!!!!!!!!!!!!!!!!!!!!!!!!! EDIT RAW FILES !!!!!!!!!!!!!!!!!!!!!!!!!
  #read and process CNV data
  cnv_data <- read.csv(filepath) %>%
    mutate(
      genotype = ifelse(grepl("B1-6548", sample), "B1/B2", sub("-.*", "", sample)), #rename samples
      sample = gsub("-", ".", sample) #rename samples
    ) %>%
    dplyr::select(cell_id, chr, state, A, B, LOH, BAF, sample, genotype, start, end) #filter columns
 
  #read in sample info
  cancer.history <- read.xlsx("~/brca-new/Williams et al./williams-samples.xlsx")
  cancer.history <- cancer.history %>%
    mutate(patient_id = gsub("-", ".", patient_id)) %>% #rename samples
    dplyr::rename(
                  sample = patient_id,  # correct direction: new = old
                  cancer = current_past_cancer.in.contralateral.breast,
                  tissue = Final.Path.Diagnosis) %>%
    dplyr::select(sample, chemo, cancer, tissue)
 
  #edit sample info
  cancer.history$chemo <- gsub("N/A", "no chemo", cancer.history$chemo)
  cancer.history$chemo <- gsub("N", "no chemo", cancer.history$chemo)
  cancer.history$chemo <- gsub("Y", "chemo", cancer.history$chemo)
  cancer.history$tissue <- gsub("R:", "", cancer.history$tissue)
  cancer.history$tissue <- gsub("R: ", "", cancer.history$tissue)
  cancer.history$tissue <- gsub("L: ", "", cancer.history$tissue)
  cancer.history$tissue <- gsub(" UDH", "", cancer.history$tissue)
  cancer.history$tissue <- gsub(" cysts", "", cancer.history$tissue)
  cancer.history$tissue <- gsub(" non-proliferative", "", cancer.history$tissue)
  cancer.history$tissue <- gsub(" ", "", cancer.history$tissue)


  #add sample info to df
  cnv_data <- cnv_data %>%
    left_join(cancer.history, by = "sample")

#!!!!!!!!!!!!!!!!!!!!!!!!! NAME SAMPLE DFS !!!!!!!!!!!!!!!!!!!!!!!!!
  #generate name
  if ("sample" %in% colnames(cnv_data)) {
    sample_name <- unique(cnv_data$sample)[1]
  } else {
    sample_name <- tools::file_path_sans_ext(basename(filepath))
  }

  #assign df and gr to global environment
  assign(paste0(sample_name, "_df"), cnv_data, envir = .GlobalEnv)

  #return sample name
  df = assign(paste0(sample_name, "_df"), cnv_data, envir = .GlobalEnv)
}

#arrange file paths
files <- c(
  # "/Users/mahadbihie/brca-new/data_directory/allele_specific_cn/B1HET43-hscn.csv",
  # "/Users/mahadbihie/brca-new/data_directory/allele_specific_cn/B1HET49-hscn.csv",
  # "/Users/mahadbihie/brca-new/data_directory/allele_specific_cn/B1HET6410-hscn.csv",
  # "/Users/mahadbihie/brca-new/data_directory/allele_specific_cn/B1HET6537-hscn.csv",
  # "/Users/mahadbihie/brca-new/data_directory/allele_specific_cn/B1HET6548-hscn.csv",
  # "/Users/mahadbihie/brca-new/data_directory/allele_specific_cn/B1HET6550-hscn.csv",
  "/Users/mahadbihie/brca-new/data_directory/allele_specific_cn/B2HET16-hscn.csv",
  "/Users/mahadbihie/brca-new/data_directory/allele_specific_cn/B2HET18-hscn.csv"
  # "/Users/mahadbihie/brca-new/data_directory/allele_specific_cn/B2HET21-hscn.csv",
  # "/Users/mahadbihie/brca-new/data_directory/allele_specific_cn/B2HET23-hscn.csv",
  # "/Users/mahadbihie/brca-new/data_directory/allele_specific_cn/B2HET25-hscn.csv",
  # "/Users/mahadbihie/brca-new/data_directory/allele_specific_cn/B2HET6532-hscn.csv",
  # "/Users/mahadbihie/brca-new/data_directory/allele_specific_cn/WT6-hscn.csv",
  # "/Users/mahadbihie/brca-new/data_directory/allele_specific_cn/WT62-hscn.csv",
  # "/Users/mahadbihie/brca-new/data_directory/allele_specific_cn/WT6752-hscn.csv"
)

#run pipeline
lapply(files, read_cnv_data)
```

### LUMI: Generate Binary Matrix of Sitka
```{r}
#format luminal CNV data
cnv.lumi <- B2.16_df %>%
  mutate(
    loci = paste(
      chr,
      format(start, scientific = FALSE, trim = TRUE),
      format(end, scientific = FALSE, trim = TRUE),
      sep = "_"
    )
  ) %>%
  group_by(loci, cell_id) %>%
  summarise(state = round(mean(state, na.rm = TRUE)), .groups = "drop") %>%
  pivot_wider(names_from = cell_id, values_from = state) %>%
  column_to_rownames(var="loci")

# Remove any leading/trailing spaces
rownames(cnv.lumi) <- trimws(rownames(cnv.lumi))

# Ensure all values are numeric
cnv.lumi[] <- lapply(cnv.lumi, as.numeric)

#save matrix for Sitka Tree
write.csv(x = cnv.lumi,
          file = "~/sitkatree/example/data/cnv_input_b2_16.csv",
          quote = FALSE,
          row.names = TRUE)
```

## *TNBC*
### TNBC: Load CNV Data
```{r}
#read in q arm cnv data
load("~/17q13q/data/q_cnv.RData")
```
### TNBC: Generate Binary Matrix of Sitka
```{r}
#format TNBC CNV data
cnv.tnbc <- q_tnbc_cnv %>%
  filter(Sample_Name == "SA501") %>%
  mutate(
    loci = paste(
      chr,
      format(start, scientific = FALSE, trim = TRUE),
      format(end, scientific = FALSE, trim = TRUE),
      sep = "_"
    )
  ) %>%
  group_by(loci, cell_id) %>%
  summarise(state = round(mean(state, na.rm = TRUE)), .groups = "drop") %>%
  pivot_wider(names_from = cell_id, values_from = state) %>%
  column_to_rownames(var="loci")

# Remove any leading/trailing spaces
rownames(cnv.tnbc) <- trimws(rownames(cnv.tnbc))

# Ensure all values are numeric
cnv.tnbc[] <- lapply(cnv.tnbc, as.numeric)

#save matrix for Sitka Tree
write.csv(x = cnv.tnbc,
          file = "~/sitkatree/example/data/cnv_input.csv",
          quote = FALSE,
          row.names = TRUE)
```

## *HGSC*
### HGSC: Load CNV Data
```{r}
#read in q arm cnv data
cnv.hgsc <- read.csv("~/17q13q/data/hgsc.cnv.csv")
```
### HGSC: Generate Binary Matrix of Sitka
```{r}
#format HGSC CNV data
cnv.hgsc <- cnv.hgsc %>%
  filter(Sample_Name == "SA501") %>%
  mutate(
    loci = paste(
      chr,
      format(start, scientific = FALSE, trim = TRUE),
      format(end, scientific = FALSE, trim = TRUE),
      sep = "_"
    )
  ) %>%
  group_by(loci, cell_id) %>%
  summarise(state = round(mean(state, na.rm = TRUE)), .groups = "drop") %>%
  pivot_wider(names_from = cell_id, values_from = state) %>%
  column_to_rownames(var="loci")

# Remove any leading/trailing spaces
rownames(cnv.hgsc) <- trimws(rownames(cnv.hgsc))

# Ensure all values are numeric
cnv.hgsc[] <- lapply(cnv.hgsc, as.numeric)

#save matrix for Sitka Tree
write.csv(x = cnv.hgsc,
          file = "~/sitkatree/example/data/cnv_input_hgsc.csv",
          quote = FALSE,
          row.names = TRUE)
```
# **SITKA Instructions (TERMINAL)**
## *SITKA Code (Terminal)*
```{r}
#Step 0: process CNV data into binary file for SITKA
Rscript cntob.R -i data/cnv_input_hgsc.csv -o data/binary.csv

# Step 1: Jitter Correction
~/sitkatree/sitka/build/install/nowellpack/bin/corrupt-straighten \
  --input ~/sitkatree/example/data/binary.csv \
  --neighborhoodSize 2
 
# Step 2: Filter Loci
~/sitkatree/sitka/build/install/nowellpack/bin/corrupt-filter --input results/all/2025-11-09-23-20-18-DnAz4OfN.exec/output.csv --lowerFraction 0.05

#Step 3: Tree Inference
JAVA_TOOL_OPTIONS="--add-opens=java.base/java.util=ALL-UNNAMED" \
~/sitkatree/sitka/build/install/nowellpack/bin/corrupt-infer-with-noisy-params \
    --model.globalParameterization true \
    --model.binaryMatrix ~/sitkatree/example/results/all/2025-11-09-23-20-46-Z8YA836d.exec/filtered.csv \
    --model.fprBound 0.1 \
    --model.fnrBound 0.5 \
    --engine PT \
    --engine.initialization FORWARD \
    --engine.nScans 10000 \
    --engine.nPassesPerScan 1 \
    --engine.nChains 10
 
#Step 4a:
~/sitkatree/sitka/build/install/nowellpack/bin/corrupt-average --csvFile results/all/2025-11-09-23-21-24-ITSiH7gv.exec/samples/phylo.csv --logisticTransform false

#Step 4b: move sample files to sample specific folder (use file path above)
cp -r ~/sitkatree/example/results/all/2025-11-09-23-21-24-ITSiH7gv.exec/samples/ ~/17q13q/my-data/hgsc-sa1052-sitka/
 
#Step 5:
  ~/sitkatree/sitka/build/install/nowellpack/bin/corrupt-greedy --tipInclusionProbabilities ReadOnlyCLMatrix results/all/2025-11-10-14-03-20-kG4d6miO.exec/average.csv

#Step 5b: move tree newick to sample specific folder (use new path to newick file)
cp ~/sitkatree/example/results/all/2025-11-10-14-25-31-tvSUW8ZH.exec/tree.newick ~/17q13q/my-data/hgsc-sa1052-sitka/
 
cp ~/sitkatree/example/results/all/2025-11-10-14-25-31-tvSUW8ZH.exec/trees.csv ~/17q13q/my-data/hgsc-sa1052-sitka/

```


# **Data Visualization**
## *B2.16: APE Visualization*
```{r}
#read the Newick file
tree <- read.tree("~/17q13q/my-data/b2.16-sitka/tree.newick")

#Step 1: Convert deletions to 1, anything else to 0
cnv_filtered <- B2.16_df %>%
  mutate(start = as.numeric(start)) %>%   #ensure numeric
  filter(
    (chr == "13" & start >= 17000000) |   #13q
    (chr == "17" & start >= 22000000)) %>%    #17q
  mutate(del = ifelse(state < 2, 1, 0)) %>% #1 = deleted, 0 = not deleted
  group_by(cell_id, chr) %>%
  summarise(
    prcnt_deleted = mean(del),   #percentage deletions across segments
    .groups = "drop") %>%
  mutate(binary = ifelse(prcnt_deleted >= 0.25, 1, 0))  #keep only cells with >=50% deletion for that chromosome


#Step 2: Pivot to wide format
cnv_wide <- cnv_filtered %>%
  pivot_wider(
    names_from = chr,
    values_from = binary,
    names_prefix = "chr"
  )

#remove NAs
cnv_wide <- cnv_wide %>%
  mutate(
    chr13 = replace_na(chr13, 0),
    chr17 = replace_na(chr17, 0)
  )

#save file
write.csv(cnv_wide, file= "~/17q13q/my-data/luminal-gBRCA2-B2.16-chr17q13q-del-cells.csv")

#Step 3: modify tree
#remove cell label on tree
tree$tip.label <- sub("^cell_", "", tree$tip.label)

#make a data frame with tip labels
tip_data <- data.frame(label = tree$tip.label) %>%
  left_join(cnv_wide, by = c("label" = "cell_id"))

#Step 4: Generate trees
#generate tree
tree.17.13 <- ggtree(tree) %<+% tip_data +  #attach metadata
  geom_tiplab(aes(color = case_when(
    chr17 == 1 ~ "17q",
    chr13 == 1 ~ "13q",
    TRUE ~ "none"
  )), size=2) +
  scale_color_manual(   name = "Chromosomal CNV",   values = c("17q" = "red", "13q" = "blue", "none" = "gray70"),   labels = c("17q" = "17q deletion", "13q" = "13q deletion", "none" = "No CNV") ) +
  theme_tree2() +
  ggtitle("B2.16: Phylogenetic Tree with chr17/13 deletions")

#print tree coded with 17q/13q deletions
tree.17.13

#save tree
ggsave(
  filename = "~/17q13q/plots/trees/B2.16-tree.png",  #or "tree_chr17_chr13.png"
  plot = tree.17.13,
  width = 17,      
  height = 17,
  dpi = 300
)

#generate tree
tree.17.13.circle <- ggtree(tree, layout="circular") %<+% tip_data +  #attach metadata
  geom_tiplab(aes(color = case_when(
    chr17 == 1 ~ "17q",
    chr13 == 1 ~ "13q",
    TRUE ~ "none"
  )), size=2) +
  scale_color_manual(   name = "Chromosomal CNV",   values = c("17q" = "red", "13q" = "blue", "none" = "gray70"),   labels = c("17q" = "17q deletion", "13q" = "13q deletion", "none" = "No CNV") ) +
  theme_tree2() +
  ggtitle("B2.16: Phylogenetic Tree with chr17/13 deletions [Circular]")

#print tree coded with 17q/13q deletions
tree.17.13.circle

#save tree
ggsave(
  filename = "~/17q13q/plots/trees/B2.16-tree-circle.png",
  plot = tree.17.13.circle,
  width = 17,      
  height = 17,    
  dpi = 300
)
```
## *B2.18: APE Visualization*
```{r}
#read the Newick file
tree <- read.tree("~/17q13q/my-data/b2.18-sitka/tree.newick")

#Step 1: Convert deletions to 1, anything else to 0
cnv_filtered <- B2.18_df %>%
  mutate(start = as.numeric(start)) %>%   #ensure numeric
  filter(
    (chr == "13" & start >= 17000000) |   #13q
    (chr == "17" & start >= 22000000)) %>%    #17q
  mutate(del = ifelse(state < 2, 1, 0)) %>% #1 = deleted, 0 = not deleted
  group_by(cell_id, chr) %>%
  summarise(
    prcnt_deleted = mean(del),   #percentage deletions across segments
    .groups = "drop") %>%
  mutate(binary = ifelse(prcnt_deleted >= 0.25, 1, 0))  #keep only cells with >=50% deletion for that chromosome

#Step 2: Pivot to wide format
cnv_wide <- cnv_filtered %>%
  pivot_wider(
    names_from = chr,
    values_from = binary,
    names_prefix = "chr"
  )

#remove NAs
cnv_wide <- cnv_wide %>%
  mutate(
    chr13 = replace_na(chr13, 0),
    chr17 = replace_na(chr17, 0)
  )

#save file
write.csv(cnv_wide, file= "~/17q13q/my-data/luminal-gBRCA2-B2.18-chr17q13q-del-cells.csv")

#Step 3: modify trees
#remove cell label on tree
tree$tip.label <- sub("^cell_", "", tree$tip.label)

#make a data frame with tip labels
tip_data <- data.frame(label = tree$tip.label) %>%
  left_join(cnv_wide, by = c("label" = "cell_id"))


#Step 4: Generate Trees
#generate tree
tree.17.13 <- ggtree(tree) %<+% tip_data +  #attach metadata
  geom_tiplab(aes(color = case_when(
    chr17 == 1 ~ "17q",
    chr13 == 1 ~ "13q",
    TRUE ~ "none"
  )), size=2) +
  scale_color_manual(   name = "Chromosomal CNV",   values = c("17q" = "red", "13q" = "blue", "none" = "gray70"),   labels = c("17q" = "17q deletion", "13q" = "13q deletion", "none" = "No CNV") ) +
  theme_tree2() +
  ggtitle("B2.18: Phylogenetic Tree with chr17/13 deletions")


#print tree coded with 17q/13q deletions
tree.17.13

#save tree
ggsave(
  filename = "~/17q13q/plots/trees/B2.18-tree.png",  #or "tree_chr17_chr13.png"
  plot = tree.17.13,
  width = 17,      
  height = 17,
  dpi = 300
)

#generate tree
tree.17.13.circle <- ggtree(tree, layout="circular") %<+% tip_data +  #attach metadata
  geom_tiplab(aes(color = case_when(
    chr17 == 1 ~ "17q",
    chr13 == 1 ~ "13q",
    TRUE ~ "none"
  )), size=2) +
  scale_color_manual(   name = "Chromosomal CNV",   values = c("17q" = "red", "13q" = "blue", "none" = "gray70"),   labels = c("17q" = "17q deletion", "13q" = "13q deletion", "none" = "No CNV") ) +
  theme_tree2() +
  ggtitle("B2.18: Phylogenetic Tree with chr17/13 deletions [Circular]")

#print tree coded with 17q/13q deletions
tree.17.13.circle

#save tree
ggsave(
  filename = "~/17q13q/plots/trees/B2.18-tree-circle.png",
  plot = tree.17.13.circle,
  width = 17,      
  height = 17,    
  dpi = 300
)
```
## *TNBC: APE Visualization*
```{r}
#read the Newick file
tree <- read.tree("C:/Users/m-bih/17q13q/my-data/tnbc-sa501-sitka/SA501.tree.newick")

#read in data frame
tnbc.cnv <- vroom("C:/Users/m-bih/brca-luminal/data/tnbc.cnv.csv")

#Step 1: Convert deletions to 1, anything else to 0
cnv_filtered_tnbc <- tnbc.cnv %>%
  mutate(start = as.numeric(start)) %>%   #ensure numeric
  filter(
    (chr == "13" & start >= 17000000) |   #13q
    (chr == "17" & start >= 22000000)) %>%    #17q
  mutate(del = ifelse(state < 2, 1, 0)) %>% #1 = deleted, 0 = not deleted
  group_by(cell_id, chr) %>%
  summarise(
    prcnt_deleted = mean(del),   #percentage deletions across segments
    .groups = "drop") %>%
  mutate(binary = ifelse(prcnt_deleted >= 0.25, 1, 0))  #keep only cells with >=50% deletion for that chromosome

#Step 2: Pivot to wide format
cnv_wide <- cnv_filtered_tnbc %>%
  #filter(prcnt_deleted >= 0.5) %>%
  pivot_wider(
    names_from = chr,
    values_from = binary,
    names_prefix = "chr"
  )

#remove NAs
cnv_wide <- cnv_wide %>%
  mutate(
    chr13 = replace_na(chr13, 0),
    chr17 = replace_na(chr17, 0)
  )

#save file
write.csv(x = cnv_wide, file = "C:/Users/m-bih/17q13q/my-data/tnbc-gBRCA1-sa501-chr17q-13q-del-cells.csv")

#Step 3: modify tree
#remove cell label on tree
tree$tip.label <- sub("^cell_", "", tree$tip.label)

#make a data frame with tip labels
tip_data <- data.frame(label = tree$tip.label) %>%
  left_join(cnv_wide, by = c("label" = "cell_id"))

#Step 4: Generate trees
#generate tree
tree.17.13 <- ggtree(tree) %<+% tip_data +  #attach metadata
  geom_tiplab(aes(color = case_when(
    chr17 == 1 ~ "17q",
    chr13 == 1 ~ "13q",
    TRUE ~ "none"
  )), size=2) +
  scale_color_manual(   name = "Chromosomal CNV",   values = c("17q" = "red", "13q" = "blue", "none" = "gray70"),   labels = c("17q" = "17q deletion", "13q" = "13q deletion", "none" = "No CNV") ) +
  theme_tree2() +
  ggtitle("TNBC-gBRCA1 (SA501): Phylogenetic Tree with chr17/13 deletions")

#print tree coded with 17q/13q deletions
tree.17.13

#save tree
ggsave(
  filename = ("C:/Users/m-bih/17q13q/plots/phylogeny plots/TNBC-gBRCA1-tree.png"),  #or "tree_chr17_chr13.png"
  plot = tree.17.13,
  width = 17,      
  height = 17, 
  dpi = 300
)

#generate tree
tree.17.13.circle <- ggtree(tree, layout="circular") %<+% tip_data +  #attach metadata
  geom_tiplab(aes(color = case_when(
    chr17 == 1 ~ "17q",
    chr13 == 1 ~ "13q",
    TRUE ~ "none"
  )), size=2) +
  scale_color_manual(   name = "Chromosomal CNV",   values = c("17q" = "red", "13q" = "blue", "none" = "gray70"),   labels = c("17q" = "17q deletion", "13q" = "13q deletion", "none" = "No CNV") ) +
  theme_tree2() +
  ggtitle("TNBC-gBRCA1 (SA1052): Phylogenetic Tree with chr17/13 deletions [Circular]")

#print tree coded with 17q/13q deletions
tree.17.13.circle

#save tree
ggsave(
  filename = "C:/Users/m-bih/17q13q/plots/phylogeny plots/TNBC-gBRCA1-tree-circle.png",
  plot = tree.17.13.circle,
  width = 17,      
  height = 17,     
  dpi = 300
)
```

## *HGSC: APE Visualization*
```{r}
#read the Newick file
tree <- read.tree("C:/Users/m-bih/17q13q/my-data/hgsc-sa1052-sitka/tree.newick")

#read in data frame
hgsc.cnv <- vroom("C:/Users/m-bih/brca-luminal/data/hgsc.cnv.csv")

#filter
sa1052 <- hgsc.cnv %>% filter(Sample_Name == "SA1052")

#Step 1: Convert deletions to 1, anything else to 0
cnv_filtered_hgsc <- sa1052 %>%
  mutate(start = as.numeric(start)) %>%   #ensure numeric
  filter(
    (chr == "13" & end >= 17000000) |    #13q → KEEP any segment touching 13q
    (chr == "17" & end >= 22000000)      #17q → KEEP any segment touching 17q
  ) %>%
  mutate(del = ifelse(state < 2, 1, 0)) %>% #1 = deleted, 0 = not deleted
  group_by(cell_id, chr) %>%
  summarise(
    prcnt_deleted = mean(del),   #percentage deletions across segments
    .groups = "drop") %>%
  mutate(binary = ifelse(prcnt_deleted >= 0.25, 1, 0))  #keep only cells with >=50% deletion for that chromosome

#Step 2: Pivot to wide format
cnv_wide <- cnv_filtered_hgsc %>%
  pivot_wider(
    names_from = chr,
    values_from = binary,
    names_prefix = "chr"
  )

#remove NAs
cnv_wide <- cnv_wide %>%
  mutate(
    chr13 = replace_na(chr13, 0),
    chr17 = replace_na(chr17, 0)
  )

#save file
write.csv(x = cnv_wide, file = "C:/Users/m-bih/17q13q/my-data/hgsc-gBRCA1-sa1052-chr17q-13q-del-cells.csv")

#Step 3: modify tree
#remove cell label on tree
tree$tip.label <- sub("^cell_", "", tree$tip.label)

#make a data frame with tip labels
tip_data <- data.frame(label = tree$tip.label) %>%
  left_join(cnv_wide, by = c("label" = "cell_id"))

#Step 4: Generate trees
#generate tree
tree.17.13 <- ggtree(tree) %<+% tip_data +  #attach metadata
  geom_tiplab(aes(color = case_when(
    chr17 == 1 ~ "17q",
    chr13 == 1 ~ "13q",
    TRUE ~ "none"
  )), size=2) +
  scale_color_manual(   name = "Chromosomal CNV",   values = c("17q" = "red", "13q" = "blue", "none" = "gray70"),   labels = c("17q" = "17q deletion", "13q" = "13q deletion", "none" = "No CNV") ) +
  theme_tree2() +
  ggtitle("HGSC-gBRCA1 (SA1052): Phylogenetic Tree with chr17/13 deletions")

#print tree coded with 17q/13q deletions
tree.17.13

#save tree
ggsave(
  filename = ("C:/Users/m-bih/17q13q/plots/phylogeny plots/HGSC-gBRCA1-tree.png"),  #or "tree_chr17_chr13.png"
  plot = tree.17.13,
  width = 17,      
  height = 17, 
  dpi = 300
)

#generate tree
tree.17.13.circle <- ggtree(tree, layout="circular") %<+% tip_data +  #attach metadata
  geom_tiplab(aes(color = case_when(
    chr17 == 1 ~ "17q",
    chr13 == 1 ~ "13q",
    TRUE ~ "none"
  )), size=2) +
  scale_color_manual(   name = "Chromosomal CNV",   values = c("17q" = "red", "13q" = "blue", "none" = "gray70"),   labels = c("17q" = "17q deletion", "13q" = "13q deletion", "none" = "No CNV") ) +
  theme_tree2() +
  ggtitle("HGSC-gBRCA1 (SA1052): Phylogenetic Tree with chr17/13 deletions [Circular]")

#print tree coded with 17q/13q deletions
tree.17.13.circle

#save tree
ggsave(
  filename = "C:/Users/m-bih/17q13q/plots/phylogeny plots/HGSC-gBRCA1-tree-circle.png",
  plot = tree.17.13.circle,
  width = 17,      
  height = 17,     
  dpi = 300
)
```

## *GENO: APE Visualization*
```{r}
#read the Newick file
tree <- read.tree("C:/Users/m-bih/17q13q/my-data/geno-sitka/tree.newick")

#read in data frame
geno_cnv <- vroom("C:/Users/m-bih/brca-luminal/data/genotype.data.cnv.csv")

#Step 1: Convert deletions to 1, anything else to 0
cnv_filtered <- geno_cnv %>%
  filter(chr %in% c("17","13")) %>%
  mutate(binary = ifelse(state < 2, 1, 0)) %>%
  dplyr::select(cell_id, chr, binary) %>%
  group_by(cell_id, chr) %>%           #group by cell and chromosome
  summarise(binary = max(binary), .groups = "drop")  #1 if any deletion, else 0

#Step 1: Convert deletions to 1, anything else to 0
cnv_filtered_geno <- geno_cnv %>%
  mutate(start = as.numeric(start)) %>%   #ensure numeric
  filter(
    (chr == "13" & start >= 17000000) |   #13q
    (chr == "17" & start >= 22000000)) %>%    #17q
  mutate(del = ifelse(state < 2, 1, 0)) %>% #1 = deleted, 0 = not deleted
  group_by(cell_id, chr) %>%
  summarise(
    prcnt_deleted = mean(del),   #percentage deletions across segments
    .groups = "drop") %>%
  mutate(binary = ifelse(prcnt_deleted >= 0.25, 1, 0))  #keep only cells with >=50% deletion for that chromosome

#save file
write.csv(x = cnv_filtered_geno, file = "C:/Users/m-bih/17q13q/my-data/184-hTERT-BRCA2-SA1188-chr17q-13q-del-cells.csv")

#Step 2: Pivot to wide format
cnv_wide <- cnv_filtered_geno %>%
  #filter(prcnt_deleted >= 0.5) %>%
  pivot_wider(
    names_from = chr,
    values_from = binary,
    names_prefix = "chr"
  )

#remove NAs
cnv_wide <- cnv_wide %>%
  mutate(
    chr13 = replace_na(chr13, 0),
    chr17 = replace_na(chr17, 0)
  )

#Step 3: modify tree
#remove cell label on tree
tree$tip.label <- sub("^cell_", "", tree$tip.label)

#make a data frame with tip labels
tip_data <- data.frame(label = tree$tip.label) %>%
  left_join(cnv_wide, by = c("label" = "cell_id"))

#Step 4: Generate trees
#generate tree
tree.17.13 <- ggtree(tree) %<+% tip_data +  #attach metadata
  geom_tiplab(aes(color = case_when(
    chr17 == 1 ~ "17q",
    chr13 == 1 ~ "13q",
    TRUE ~ "none"
  )), size=2) +
  scale_color_manual(   name = "Chromosomal CNV",   values = c("17q" = "red", "13q" = "blue", "none" = "gray70"),   labels = c("17q" = "17q deletion", "13q" = "13q deletion", "none" = "No CNV") ) +
  theme_tree2() +
  ggtitle("GENO-gBRCA1 (SA501): Phylogenetic Tree with 25% chr17/13 deletions")

#print tree coded with 17q/13q deletions
tree.17.13

#save tree
ggsave(
  filename = ("C:/Users/m-bih/17q13q/plots/phylogeny plots/GENO-gBRCA1-tree-25.png"),  #or "tree_chr17_chr13.png"
  plot = tree.17.13,
  width = 17,      
  height = 17, 
  dpi = 300
)

#generate tree
tree.17.13.circle <- ggtree(tree, layout="circular") %<+% tip_data +  #attach metadata
  geom_tiplab(aes(color = case_when(
    chr17 == 1 ~ "17q",
    chr13 == 1 ~ "13q",
    TRUE ~ "none"
  )), size=2) +
  scale_color_manual(   name = "Chromosomal CNV",   values = c("17q" = "red", "13q" = "blue", "none" = "gray70"),   labels = c("17q" = "17q deletion", "13q" = "13q deletion", "none" = "No CNV") ) +
  theme_tree2() +
  ggtitle("GENO-gBRCA1 (SA1052): Phylogenetic Tree with chr17/13 deletions [Circular]")

#print tree coded with 17q/13q deletions
tree.17.13.circle

#save tree
ggsave(
  filename = "C:/Users/m-bih/17q13q/plots/phylogeny plots/GENO-gBRCA1-tree-circle.png",
  plot = tree.17.13.circle,
  width = 17,      
  height = 17,     
  dpi = 300
)
```
